components/plurals/src/lib.rs

```diff
@@ -341,6 +343,38 @@ impl PluralRules {
         self.selector.select(&input.into())
     }
 
+    /// Returns all [`Plural Categories`] appropriate for a [`PluralRules`] object
+    /// based on the [`LanguageIdentifier`] and [`PluralRuleType`].
+    ///
+    /// The [`Plural Categories`] are returned in alphabetically sorted order.
+    ///
+    /// The category [`PluralCategory::Other`] is always included.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu::locid::macros::langid;
+    /// use icu::plurals::{PluralRules, PluralRuleType, PluralCategory};
+    /// use icu_provider::inv::InvariantDataProvider;
+    ///
+    /// let lid = langid!("ja");
```

Comment:
Suggestion (optional): Choose a more interesting locale than `"ja"` for the example.  Something that has at least 3 plural forms.

Reply:
I chose `"ja"` because it only has `"other"` and I don't think that `InvariantDataProvider` can actually give me real data. 

I think that a locale that supports only `"other"` would be the only thing that I could reasonably put in an example such as this, unless we just drop the assertions and show how to call it without actually verifying anything. 

It's a pretty poor assertion in the first place, so I wouldn't be against having no assertions in this example. if `"ja"` were to ever get different rules, this test would continue to pass (incorrectly so) since the `InvariantDataProvider` is only capable of providing `"other"`.

Reply:
I've updated the example to just collect the categories into a `Vec` and assert that `Other` is present, which is true for all langids.